### PR TITLE
AM-581-JUnit-code-coverage_new

### PR DESCRIPTION
**Pull request by nitinsahu1**
JIRA link (if applicable)
https://tools.hmcts.net/jira/browse/AM-581

Change description
JUnit code coverage : Achieve minimum code coverage as per standard

Does this PR introduce a breaking change? (check one with "x")

[ ] Yes
[x ] No

### CODE REVIEW

File: src/test/java/uk/gov/hmcts/reform/ccd/document/am/controller/WelcomeTest.java [hunk 1]
```diff
@@ -17,7 +20,22 @@ public void shouldReturnWelcomeMessage() {
...
+
+    @Test
+    public void shouldGetReturnIndex() {
+        String response = caseDocumentAmController.index();
+        assertNotNull(response, "Valid Response from WelcomeController");
+        assertEquals(response, "redirect:swagger-ui.html", "Response Value");
     }
+
+    @Test
+    public void shouldGetReturnIndex1() {
```
Comment (shrivsk) [inline src/test/java/uk/gov/hmcts/reform/ccd/document/am/controller/WelcomeTest.java]: Methode name should be more specific if possible

File: src/test/java/uk/gov/hmcts/reform/ccd/document/am/controller/advice/CaseDocumentControllerAdviceTest.java [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+            welcomeController.getException("badRequestException");
+        });
+    }
+
+    @Test
+    void testCaseNotFoundException() {
+        Assertions.assertThrows(CaseNotFoundException.class, () -> {
+            welcomeController.getException("caseNotFoundException");
+        });
+    }
```
Comment (shrivsk) [inline src/test/java/uk/gov/hmcts/reform/ccd/document/am/controller/advice/CaseDocumentControllerAdviceTest.java]: is it possible to add some more test like 
1> Type mismatch exception 
2> Boundary value exception etc

File: build.gradle [hunk 3]
```diff
@@ -179,15 +179,12 @@ project.tasks['pitest'].group="verification"
 
 sonarqube {
   properties {
-    property "sonar.projectName", "Reform :: ccd-case-document-am-api"
```
Comment (aashishaggarwal123) [inline build.gradle]: Please disable the sonar for now, as we cannot configure the limit and it will be enabled once coverage reaches 80%

File: src/main/java/uk/gov/hmcts/reform/ccd/document/am/controller/WelcomeController.java [hunk 4]
```diff
@@ -1,18 +1,20 @@
...
-
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 import org.springframework.http.ResponseEntity;
+import org.springframework.http.converter.HttpMessageConversionException;
 import org.springframework.web.bind.annotation.GetMapping;
 import org.springframework.web.bind.annotation.PathVariable;
 import org.springframework.web.bind.annotation.RequestMapping;
 import org.springframework.web.bind.annotation.RestController;
+import uk.gov.hmcts.reform.ccd.document.am.controller.advice.exception.*;
```
Comment (aashishaggarwal123) [inline src/main/java/uk/gov/hmcts/reform/ccd/document/am/controller/WelcomeController.java]: Kindly disable your intellij import optimise and remove the * based imported classes.

File: src/test/java/uk/gov/hmcts/reform/ccd/document/am/controller/WelcomeTest.java (hunk 1 above)
Comment (nitinsahu1) [reply src/test/java/uk/gov/hmcts/reform/ccd/document/am/controller/WelcomeTest.java]: Done

File: src/test/java/uk/gov/hmcts/reform/ccd/document/am/controller/advice/CaseDocumentControllerAdviceTest.java (hunk 2 above)
Comment (nitinsahu1) [reply src/test/java/uk/gov/hmcts/reform/ccd/document/am/controller/advice/CaseDocumentControllerAdviceTest.java]: As discussed this is a generic class for handling exceptions and would be updated gradually based on the new scenarios.

File: build.gradle (hunk 3 above)
Comment (nitinsahu1) [reply build.gradle]: Done

File: src/main/java/uk/gov/hmcts/reform/ccd/document/am/controller/WelcomeController.java (hunk 4 above)
Comment (nitinsahu1) [reply src/main/java/uk/gov/hmcts/reform/ccd/document/am/controller/WelcomeController.java]: Done
